src/engine/engineworker.cpp

```diff
@@ -16,12 +16,17 @@ void EngineWorker::run() {
 
 void EngineWorker::setScheduler(EngineWorkerScheduler* pScheduler) {
     m_pScheduler = pScheduler;
```

Comment:
We should at least check by a debug assertion that the worker currently doesn't have any scheduler when this function is called. Otherwise it would be scheduled twice.